include/engine/hdf5/HDF5WriterP.h

```diff
@@ -0,0 +1,224 @@
+  void Write(Variable<long int> &variable, const long int *values);
+  void Write(Variable<unsigned long int> &variable,
+             const unsigned long int *values);
+  void Write(Variable<long long int> &variable, const long long int *values);
+  void Write(Variable<unsigned long long int> &variable,
+             const unsigned long long int *values);
+  void Write(Variable<float> &variable, const float *values);
+  void Write(Variable<double> &variable, const double *values);
+  void Write(Variable<long double> &variable, const long double *values);
+  void Write(Variable<std::complex<float>> &variable,
+             const std::complex<float> *values);
+  void Write(Variable<std::complex<double>> &variable,
+             const std::complex<double> *values);
+  void Write(Variable<std::complex<long double>> &variable,
+             const std::complex<long double> *values);
+
+  void Write(const std::string variableName, const char *values);
+  void Write(const std::string variableName, const unsigned char *values);
+  void Write(const std::string variableName, const short *values);
+  void Write(const std::string variableName, const unsigned short *values);
+  void Write(const std::string variableName, const int *values);
+  void Write(const std::string variableName, const unsigned int *values);
+  void Write(const std::string variableName, const long int *values);
+  void Write(const std::string variableName, const unsigned long int *values);
+  void Write(const std::string variableName, const long long int *values);
+  void Write(const std::string variableName,
+             const unsigned long long int *values);
+  void Write(const std::string variableName, const float *values);
+  void Write(const std::string variableName, const double *values);
+  void Write(const std::string variableName, const long double *values);
+  void Write(const std::string variableName, const std::complex<float> *values);
+  void Write(const std::string variableName,
+             const std::complex<double> *values);
+  void Write(const std::string variableName,
+             const std::complex<long double> *values);
+
+  void Close(const int transportIndex = -1);
+  /*
+  {
+      std::cout<<" ===> CLOSING HDF5 <===== "<<std::endl;
+      //H5Dclose(_dset_id);
+      //H5Sclose(_filespace);
+      //H5Sclose(_memspace);
+      //H5Pclose(_plist_id);
+      H5Fclose(_file_id);
+  }
+  */
+private:
+  capsule::STLVector
+      m_Buffer; ///< heap capsule, contains data and metadata buffers
+
+  void Init();
+  void clean();
+
+  hid_t _plist_id, _file_id, _dset_id;
+  hid_t _memspace, _filespace;
+
+  hid_t DefH5T_COMPLEX_DOUBLE;
+  hid_t DefH5T_COMPLEX_FLOAT;
+  hid_t DefH5T_COMPLEX_LongDOUBLE;
+
+  hid_t DefH5T_filetype_COMPLEX_DOUBLE;
+  hid_t DefH5T_filetype_COMPLEX_FLOAT;
+  hid_t DefH5T_filetype_COMPLEX_LongDOUBLE;
+
+  template <class T>
+  void UseHDFWrite(Variable<T> &variable, const T *values, hid_t h5type);
+
+  /*
```

Comment:
It looks like the implementation ahs already been moved to the cpp file so the large commented out block is just dead code.  Please delete.